src/Docker/Image/Builder/ImageBuilder.php

```diff
@@ -91,17 +86,20 @@ class ImageBuilder extends Image
      */
     private $parentImage;
 
-    public function __construct(ObjectEncryptor $encryptor, Component $component, LoggerInterface $logger)
+    private JobScopedEncryptor $jobScopedEncryptor;
+
+    public function __construct(JobScopedEncryptor $jobScopedEncryptor, Component $component, LoggerInterface $logger)
     {
-        parent::__construct($encryptor, $component, $logger);
+        parent::__construct($component, $logger);
+        $this->jobScopedEncryptor = $jobScopedEncryptor;
+
         $config = $component->getImageDefinition();
         if (isset($config["build_options"])) {
             if (isset($config["build_options"]["repository"]["username"])) {
                 $this->repoUsername = $config["build_options"]["repository"]["username"];
             }
             if (isset($config["build_options"]["repository"]["#password"])) {
-                $this->repoPassword =
-                    $this->getEncryptor()->decrypt($config["build_options"]["repository"]["#password"]);
+                $this->repoPassword = $this->jobScopedEncryptor->decrypt($config["build_options"]["repository"]["#password"]);
```

Comment:
Abych se drzel stejnyho pristupu, jak byl driv, hodil jsem sem ten `JobScopedEncryptor` (predtim se to normalne decryptovalo tin encryptorem od jobu), ale prijde mi to divny.

Reply:
ok